llvm/test/CodeGen/ARM/srem-seteq-illegal-types.ll

```diff
@@ -244,8 +243,9 @@ define i1 @test_srem_pow2_setne(i6 %X) nounwind {
 ;
 ; ARM8-LABEL: test_srem_pow2_setne:
 ; ARM8:       @ %bb.0:
-; ARM8-NEXT:    sbfx r1, r0, #0, #6
-; ARM8-NEXT:    ubfx r1, r1, #9, #2
+; ARM8-NEXT:    lsl r1, r0, #26
```

Comment:
Looks like ARM prefers to fold `asr #31` over folding an immediate. AArch64 on the other has a `let AddComplexity = 6` on the ANDWri pattern to priorite the immediate.

Reply:
Considering these get optimized very differently by opt, and I don't see this coming up elsewhere I think it is OK. (I may look into it in the future to see if there is something we can do more generally to improve it).

Reply:
Thanks @davemgreen